### ISSUE TITLE
Base config doesn't overwrite run config when continuing run

### ISSUE THREAD

**Issue by Jubeku** [ecmwf/WeatherGenerator#1663]
### What happened?

The only way currently to load a given checkpoint is by continuing the respective run through `train_continue`. 
Now if you want to load and freeze encoder weights and fine-tune a new decoder (or forecast engine), you need to `train_continue` the pre-trained model, and overwrite the config regarding the new decoder (or forecast engine) architecture. For this you should be able to specify a base-config.
However, currently the base config is only loaded if `from_run_id == None`: https://github.com/ecmwf/WeatherGenerator/blob/develop/packages/common/src/weathergen/common/config.py#L366-L372

### What are the steps to reproduce the bug?

Run in interactive mode: 
`uv run train_continue --from_run_id <run_id> --base_config <path to config>`

**Comment by Jubeku** [comment]
One option is to use `--config` instead to overwrite previous parameters. But there is a complication with the new dict-like config structure with the result that merging will simply append dictionaries instead of overwriting them. 
One example case I had is that the model checkpoint was trained with physical loss and now I want to use the encoder for a diffusion training with latent loss. Specifying the new `--config` I end up with both loss terms:
`'losses': {'physical': {'type': 'LossPhysical', 'loss_fcts': {'mse': {}}}, 'latent_diff': {'type': 'LossLatentDiffusion', 'weight': 1.0, 'target_and_aux_calc': 'DiffusionLatentTargetEncoder', 'loss_fcts': {'mse': {}}}}`

**Comment by clessig** [comment]
> se `--config` instead to overwrite previous parameters. But there is a complication with the new dict-like config structure with the result that merging will simply append dictionaries instead of overwriting them.
> One example case I had is that the model checkpoint was trained with physical loss and now I want to use the encoder for a diffusion training with latent loss.

That's why we introduced the `enabled` flag to `losses`, `model_inputs` and `target_input` (we can add to other parts). So you can disable the physical loss using `enabled: False`.

**Comment by Jubeku** [comment]
> > se `--config` instead to overwrite previous parameters. But there is a complication with the new dict-like config structure with the result that merging will simply append dictionaries instead of overwriting them.
> > One example case I had is that the model checkpoint was trained with physical loss and now I want to use the encoder for a diffusion training with latent loss.
> 
> That's why we introduced the `enabled` flag to `losses`, `model_inputs` and `target_input` (we can add to other parts). So you can disable the physical loss using `enabled: False`.

I see, so I need the full config from pre-training and disable parts of it. Let me give this a try.

**Comment by clessig** [comment]
> > > se `--config` instead to overwrite previous parameters. But there is a complication with the new dict-like config structure with the result that merging will simply append dictionaries instead of overwriting them.
> > > One example case I had is that the model checkpoint was trained with physical loss and now I want to use the encoder for a diffusion training with latent loss.
> > 
> > 
> > That's why we introduced the `enabled` flag to `losses`, `model_inputs` and `target_input` (we can add to other parts). So you can disable the physical loss using `enabled: False`.
> 
> I see, so I need the full config from pre-training and disable parts of it. Let me give this a try.

I think 

```
losses : { 
  "same_name" : {
    type: physical,
    enabled: False,
  }
}
```

should work. But would also need to test

**Comment by grassesi** [comment]
@Jubeku does it work for you with the setup Christian suggested? I will close this issue for now. Feel free to reopen it again if you still run into issues 👍
